homeassistant/helpers/selector.py

```diff
@@ -295,25 +300,179 @@ class StringSelector(Selector):
+            )
+        }
     )
 
     def __call__(self, data: Any) -> Any:
         """Validate the passed selection."""
-        selected_option = vol.In(self.config["options"])(cv.string(data))
-        return selected_option
+        if isinstance(self.config["options"][0], str):
+            options = self.config["options"]
+        else:
+            options = [option["value"] for option in self.config["options"]]
+        return vol.In(options)(cv.string(data))
+
+
+@SELECTORS.register("attribute")
+class AttributeSelector(Selector):
+    """Selector for an entity attribute."""
+
+    selector_type = "attribute"
+
+    CONFIG_SCHEMA = vol.Schema({vol.Required("entity_id"): cv.entity_id})
+
+    def __call__(self, data: Any) -> str:
+        """Validate the passed selection."""
+        return cv.string(data)
+
+
+@SELECTORS.register("duration")
+class DurationSelector(Selector):
+    """Selector for a duration."""
+
+    selector_type = "duration"
+
+    CONFIG_SCHEMA = vol.Schema({})
+
+    def __call__(self, data: Any) -> timedelta:
+        """Validate the passed selection."""
+        duration: timedelta = cv.time_period_dict(data)
+        return duration
+
+
+@SELECTORS.register("icon")
+class IconSelector(Selector):
+    """Selector for an icon."""
+
+    selector_type = "icon"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {vol.Optional("placeholder"): str, vol.Optional("fallbackPath"): str}
+    )
+
+    def __call__(self, data: Any) -> str:
+        """Validate the passed selection."""
+        return cv.string(data)
+
+
+@SELECTORS.register("theme")
+class ThemeSelector(Selector):
+    """Selector for an theme."""
+
+    selector_type = "theme"
+
+    CONFIG_SCHEMA = vol.Schema({})
+
+    def __call__(self, data: Any) -> str:
+        """Validate the passed selection."""
+        return cv.string(data)
+
+
+@SELECTORS.register("media")
+class MediaSelector(Selector):
+    """Selector for media."""
+
+    selector_type = "media"
+
+    CONFIG_SCHEMA = vol.Schema({})
+    DATA_SCHEMA = vol.Schema(
+        {
+            vol.Optional("entity_id"): cv.entity_id_or_uuid,
+            vol.Optional("media_content_id"): str,
+            vol.Optional("media_content_type"): str,
+            vol.Optional("metadata"): vol.Schema(
+                {
+                    vol.Optional("title"): str,
+                    vol.Optional("thumbnail"): vol.Any(str, None),
+                    vol.Optional("media_class"): str,
+                    vol.Optional("children_media_class"): vol.Any(str, None),
+                    vol.Optional("navigateIds"): [
+                        vol.Schema(
+                            {
+                                vol.Required("media_content_type"): str,
+                                vol.Required("media_content_id"): str,
+                            }
+                        )
+                    ],
+                }
+            ),
+        }
+    )
```

Comment:
Everything is optional, so an empty `dict` is a valid selection. Is that intentional?

Reply:
it is needed in the frontend, but it is not valid. It needs `entity_id`, `media_content_id` and `media_content_type`